generate-macos-wrapper.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+set -e
+
+if test "$#" -ne 3
+then
+	echo >&2 "usage: $0 <OUTPUT> <BUILD_OPTIONS> <VERSION_FILE>"
+	exit 1
+fi
+
+OUTPUT="$1"
+BUILD_OPTIONS="$2"
+VERSION_FILE="$3"
+
+. "$BUILD_OPTIONS"
+. "$VERSION_FILE"
+
+rm -f "$OUTPUT" "$OUTPUT"+
+
+cat >"$OUTPUT"+ <<EOF
+#!$SHELL_PATH
+
+if test "z\$*" = zversion ||
+   test "z\$*" = z--version
+then
+	echo 'git-gui version $GITGUI_VERSION'
+else
+	libdir="\${GIT_GUI_LIB_DIR:-$GITGUI_LIBDIR}"
+	exec "\$libdir/Git Gui.app/Contents/MacOS/$TKEXECUTABLE" "\$0" "\$@"
```

Comment:
We have quite a few levels of quoting here. The values of `$GITGUI_LIBDIR` and `$TKEXECUTABLE` are taken from the file `$BUILD_OPTIONS`, but are not protected against meta-characters that could be interpreted as part of the shell syntax.

I suggest to construct the file by including the contents of `$BUILD_OPTIONS` and `$VERSION_FILE`:

```sh
(
   echo "#!$SHELL_PATH"
   cat "$BUILD_OPTIONS" "$VERSION_FILE" - <<-'EOF'
   ... rest as above without the extra quoting plus dq around "$GITGUI_VERSION"
   EOF
) >"$OUTPUT"
```

Reply:
That reads much better indeed, thanks!